pkg/events/derive/net_flow_tcp.go

```diff
@@ -0,0 +1,154 @@
+package derive
+
+import (
+	"net"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+
+	"github.com/aquasecurity/tracee/pkg/dnscache"
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+func NetFlowTCPBegin(cache *dnscache.DNSCache) DeriveFunction {
+	return deriveSingleEvent(events.NetFlowTCPBegin, deriveNetFlowTCPBeginArgs(cache))
+}
+
+func deriveNetFlowTCPBeginArgs(cache *dnscache.DNSCache) deriveArgsFunction {
+	return func(event trace.Event) ([]interface{}, error) {
+		ret := event.ReturnValue
+
+		pktDirection := getPacketDirection(&event)
+		ingress := pktDirection == trace.PacketIngress
+		egress := pktDirection == trace.PacketEgress
+
+		begin := ret&flowTCPBegin == flowTCPBegin
+		end := ret&flowTCPEnd == flowTCPEnd
+
+		ipv4 := ret&familyIpv4 == familyIpv4
+		ipv6 := ret&familyIpv6 == familyIpv6
+
+		// Return fast if not the proper event (egress/ingress, begin/end).
+
+		if !begin && !end {
+			logger.Debugw("not a TCP flow event", "id", event.EventID)
+			return nil, nil
+		}
+		if !ingress && !egress {
+			logger.Debugw("wrong flow direction", "id", event.EventID)
+			return nil, nil
+		}
+		if !ipv4 && !ipv6 {
+			logger.Debugw("base layer type not supported", "id", event.EventID)
+			return nil, nil
+		}
+
+		// Proper event, now parse the packet.
+
+		var srcIP, dstIP net.IP
```

Comment:
This part is basically copied from the tcp packet code right? Maybe move to a common function to deduplicate?

Reply:
Not only that, there is a big refactor that can be made among all packet parsing derivation functions. Didn't start doing it yet, but thought about it. Let me tackle that after this PR, not during.

Reply:
Done.